tornado/tcpclient.py

```diff
@@ -93,7 +94,13 @@ def try_connect(self, addrs):
                 self.future.set_exception(self.last_error or
                                           IOError("connection failed"))
             return
-        future = self.connect(af, addr)
+        try:
+            future = self.connect(af, addr)
+        except socket.error as e:
```

Comment:
But it can't go there. The error is raised before the future is created (from `_create_stream` which will fail to create an IPv6 socket).